pySDC/core/Lagrange.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+from scipy.special import roots_legendre
+
+
+def computeFejerRule(n):
+    # Fejer rule of the first kind
+    # Computation using DFT (Waldvogel 2006)
+    # Inspired from quadpy (https://github.com/nschloe/quadpy @Nico_Schlömer)
+
+    # Initialize output variables
+    n = int(n)
+    nodes = np.empty(n, dtype=float)
+    weights = np.empty(n, dtype=float)
+
+    # Compute nodes
+    theta = np.arange(1, n + 1, dtype=float)[-1::-1]
+    theta *= 2
+    theta -= 1
+    theta *= np.pi / (2 * n)
+    np.cos(theta, out=nodes)
+
+    # Compute weights
+    # -- Initial variables
+    N = np.arange(1, n, 2)
+    lN = len(N)
+    m = n - lN
+    K = np.arange(m)
+    # -- Build v0
+    v0 = np.concatenate([
+        2 * np.exp(1j * np.pi * K / n) / (1 - 4 * K**2),
+        np.zeros(lN + 1)])
+    # -- Build v1 from v0
+    v1 = np.empty(len(v0) - 1, dtype=complex)
+    np.conjugate(v0[:0:-1], out=v1)
+    v1 += v0[:-1]
+    # -- Compute inverse fourier transform
+    w = np.fft.ifft(v1)
+    if max(w.imag) > 1.0e-15:
+        raise ValueError(
+            f'Max imaginary value to important for ifft: {max(w.imag)}')
+    # -- Store weights
+    weights[:] = w.real
+
+    return nodes, weights
+
+
+class LagrangeApproximation(object):
+
```

Comment:
If this should be part of the `core` part of pySDC, I'd require a lot more documentation here and everywhere. In the current state I would prefer to let this be part of the playground.

Reply:
Done